packages/async/src/babel-plugin.ts

```diff
@@ -0,0 +1,143 @@
+import * as Types from '@babel/types';
+import {NodePath, Binding} from '@babel/traverse';
+
+const DEFAULT_PROCESS_PACKAGES = {
+  '@shopify/react-async': ['createAsyncComponent'],
+  '@shopify/react-graphql': ['createAsyncQueryComponent'],
+};
+
+interface State {
+  processPackages?: Map<string, string[]>;
+  opts?: {packages?: {[key: string]: string[]}};
+}
+
+export default function asyncBabelPlugin({types: t}: {types: typeof Types}) {
```

Comment:
I had some [tests](https://github.com/Shopify/quilt/blob/678536dd2ef0ab7817687ed5026f2e19473dc3ed/packages/async-chunks/src/lib/babel/tests/index.test.tsx) from when I worked on the babel plugin. Might be worth pulling some of those in.